File: book-catalogue/src/main/java/com/example/bookcatalogue/model/Book.java
package com.example.bookcatalogue.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;
import java.sql.Date;


@Entity
@Table(name = "BOOK")
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Book  extends BaseEntity{

    @Column(nullable = false, unique = false)
    private String title;

    @Column(nullable = false, unique = false)
    private String genre;

    @Column(nullable = false, unique = false)
    private Date publicationDate;
}

File: ng-bookcatalogue/src/app/classes/book.ts
import { Author } from './author';

export class Book {
  id: number;
  title: string;
  genre: string;
  publicationDate: string;
  authors: Author[];

  public constructor(id: number,
                     title?: string,
                     genre?: string,
                     pdate?: string,
                     authors?: Author[]) {
    this.id = id;
    this.title = title;
    this.genre = genre;
    this.publicationDate = pdate;
    this.authors = authors;
  }
}
